src/View.Uno.Shared/Extensions/TextBox/FormattingTextBoxBehavior.iOS.cs

```diff
@@ -0,0 +1,156 @@
+			if (nativeView == null)
+			{
+				if (typeof(FormattingTextBoxBehavior).Log().IsEnabled(LogLevel.Error))
+				{
+					if ((textbox as DependencyObject).FindFirstChild<MultilineTextBoxView>() != null)
+					{
+						typeof(FormattingTextBoxBehavior).Log().Error($"`{nameof(MultilineTextBoxView)}` is not supported. Make sure `AcceptsReturn` is set to false and `TextWrapping` is `NoWrap`.");
+					}
+					else
+					{
+						typeof(FormattingTextBoxBehavior).Log().Error($"Could not find a native view of type `{nameof(SinglelineTextBoxView)}`.");
+					}
+				}
+
+				return;
+			}
+
+			// enabled and format may be null because the value may be outdated or not provided
+
+			if (enabled ?? GetIsEnabled(textbox))
+			{
+				UITextFieldDelegatingDelegate.DetourOrUpdate(nativeView, format ?? GetTextFormat(textbox));
+			}
+			else
+			{
+				(nativeView.Delegate as UITextFieldDelegatingDelegate)?.RestoreDefault();
+			}
+		}
+
+		public class UITextFieldDelegatingDelegate : UITextFieldDelegate
+		{
+			private SinglelineTextBoxView _nativeView;
+			private IUITextFieldDelegate _delegate;
+
+			public string TextFormat { get; set; }
+
+			public UITextFieldDelegatingDelegate(SinglelineTextBoxView nativeView)
+			{
+				this._nativeView = nativeView;
+				this._delegate = nativeView.Delegate;
+			}
+
+			public static UITextFieldDelegatingDelegate DetourOrUpdate(SinglelineTextBoxView nativeView, string format)
+			{
+				if (nativeView.Delegate is UITextFieldDelegatingDelegate @delegate)
+				{
+					@delegate.TextFormat = format;
+					return @delegate;
+				}
+				else if (nativeView.Delegate is SinglelineTextBoxDelegate)
+				{
+					var result = new UITextFieldDelegatingDelegate(nativeView);
+					result.TextFormat = format;
+					nativeView.Delegate = result;
+
+					return result;
+				}
+				else
+				{
+					throw new InvalidOperationException();
+				}
+			}
+
+			public void RestoreDefault()
+			{
+				_nativeView.Delegate = _delegate;
+				_delegate = null;
+				_nativeView = null;
+
+				Dispose(true);
+			}
+
+			public override bool ShouldChangeCharacters(UITextField textField, NSRange range, string replacementString)
+			{
+				// backspacing / deleting
+				if (string.IsNullOrEmpty(replacementString))
+				{
+					return DefaultBehavior();
+				}
+
+				var result = ValidateInput(TextFormat, (int)range.Location, replacementString);
+				result &= DefaultBehavior();
+
+				return result;
+
+				bool DefaultBehavior() => _delegate.ShouldChangeCharacters(textField, range, replacementString);
+			}
+
+
+			#region restore uno implementations
+			//public override bool ShouldChangeCharacters(UITextField textField, NSRange range, string replacementString) => _delegate.ShouldChangeCharacters(textField, range, replacementString);
+
+			public override bool ShouldReturn(UITextField textField) => _delegate.ShouldReturn(textField);
+
+			public override bool ShouldBeginEditing(UITextField textField) => _delegate.ShouldBeginEditing(textField);
```

Comment:
Remove this, it's causing crashes, had to do a custom implementation in one project.

Reply:
I'll comment them out, do these `EditingStarted` `EditingEnded` have issues too?

Reply:
I don't think so, just remove ShouldBeginEditing and it should be good.